src/precice/impl/ValidationMacros.hpp

```diff
@@ -78,53 +97,76 @@
-                                                        << _accessorName << "\" by defining <write-data mesh=\"" << context.mesh->getName() << "\" name=\"" << context.getName() << "\" />.");
-
-/// Validates a given dataID
+#define PRECICE_REQUIRE_DATA_WRITE_IMPL(id)                                                                                      \
+  PRECICE_VALIDATE_DATA_ID_IMPL(id)                                                                                              \
+  PRECICE_CHECK((_accessor->isDataUsed(id) && _accessor->isDataWrite(id)),                                                       \
+                "This participant does not use Data \"{0}\", but attempted to write it. "                                        \
+                "Please extend the configuarion of partiticipant \"{1}\" by defining <write-data mesh=\"{2}\" name=\"{0}\" />.", \
+                _accessor->getDataName(id), _accessorName, _accessor->getMeshNameFromData(id));
+
+/** Validates a given dataID
+ * This macros creates the "id" in a local scope and provides it to the called implementation.
+ */
 #define PRECICE_VALIDATE_DATA_ID(dataID) \
   do {                                   \
     const auto id = (dataID);            \
     PRECICE_VALIDATE_DATA_ID_IMPL(id)    \
   } while (false)
 
-/// Validates a dataID and checks for read access
+/** Validates a given dataID and checks for read access
+ * This macros creates the "id" in a local scope and provides it to the called implementation.
+ */
 #define PRECICE_REQUIRE_DATA_READ(dataID) \
   do {                                    \
     const auto id = (dataID);             \
     PRECICE_REQUIRE_DATA_READ_IMPL(id)    \
   } while (false)
 
-/// Validates a dataID and checks for write access
+/** Validates a given dataID and checks for write access
+ * This macros creates the "id" in a local scope and provides it to the called implementation.
+ */
 #define PRECICE_REQUIRE_DATA_WRITE(dataID) \
   do {                                     \
     const auto id = (dataID);              \
     PRECICE_REQUIRE_DATA_WRITE_IMPL(id)    \
   } while (false)
+
+//
+// DATA VALUE VALIDATION
+//
+#ifdef NDEBUG
+
+#define PRECICE_VALIDATE_DATA(data, size) \
+  {                                       \
+  }
+#else //NDEBUG
+
+#define PRECICE_VALIDATE_DATA(data, size) \
+  PRECICE_CHECK(std::all_of(data, data + size, [](double val) { return std::isfinite(val); }), "One of the given data values is either plus or minus infinity or NaN.");
+
+#endif
+
+#define PRECICE_EXPERIMENTAL_API()                                                                                                                 \
+  PRECICE_CHECK(_allowsExperimental, "You called the API function \"{}\", which is part of the experimental API. "                                 \
+                                     "You may unlock the full API by specifying <solver-interface experimental=\"true\" /> in the configuration. " \
```

Comment:
The `solver-interface` tag has no backslash at the end and you might want to consider adding the `dimension=..` here as well, because it is required in this tag anyway. Otherwise it might be confused with the very first tag.

Reply:
Thanks, this is indeed incorrect.

I'll change this to `<solver-interface experimental=\"true\" ... >`, this way we don't need to update it in case there are future changes.

Reply:
Done